src/routing_table/mod.rs

```diff
@@ -385,6 +385,47 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         }
     }
 
+    /// Are we among the `count` closest nodes to `name`?
+    pub fn is_closest(&self, name: &T, count: usize) -> bool {
+        self.closest_names(name, count).is_some()
+    }
+
+    /// Returns the `count` closest entries to `name` in the routing table, including our own name,
+    /// sorted by ascending distance to `name`. If we are not close, returns `None`.
+    pub fn closest_names(&self, name: &T, count: usize) -> Option<Vec<&T>> {
+        let result = self.groups
```

Comment:
Yes, true.  I did look through itertools for an equivalent of [`partial_sort`](http://en.cppreference.com/w/cpp/algorithm/partial_sort) but couldn't find it.  I guess in this case I feel the decrease in code complexity is worth the efficiency hit given that we've not profiled the code yet.  But happy to go with the flow.